chia/full_node/mempool_check_conditions.py

```diff
@@ -164,10 +163,7 @@ def get_name_puzzle_conditions(
 def get_puzzle_and_solution_for_coin(generator: BlockGenerator, coin_name: bytes, max_cost: int):
     try:
         block_program = generator.program
-        if not generator.generator_args:
```

Comment:
this check confused me. `generator_args` is not `Optional`

Reply:
I think when this was written, there was not an associated type - I think there was code in the block parsing code that might have converted empty list -> None